[OData] Patch On Entity Results In NPE

**Describe the bug**
When executing a PATCH on an existing entity to only update a single property, e.g.

```
PATCH <baseUrl>/Domain('ABC')
Body: { "Name": "NewName"}
````
results in an NPE:
```
Caused by: org.apache.olingo.odata2.core.ep.EntityProviderProducerException: An exception of type 'NullPointerException' occurred.
	at org.apache.olingo.odata2.core.ep.JsonEntityProvider.writeEntry(JsonEntityProvider.java:155) ~[olingo-odata2-core-3.0.0-...]
	at org.apache.olingo.odata2.core.ep.ProviderFacadeImpl.writeEntry(ProviderFacadeImpl.java:129) ~[olingo-odata2-core-3.0.0-...]
	at org.apache.olingo.odata2.api.ep.EntityProvider.writeEntry(EntityProvider.java:623) ~[olingo-odata2-api-3.0.0-sap-17.jar:3.0.0-sap-17]
	at org.eclipse.dirigible.engine.odata2.sql.processor.AbstractSQLProcessor.updateEntity(AbstractSQLProcessor.java:597) ~[dirigible-odata-core-5.12.9.jar:na]
	at org.eclipse.dirigible.engine.odata2.sql.processor.AbstractSQLProcessor$$FastClassBySpringCGLIB$$f667424d.invoke(<generated>) ~[dirigible-odata-core-5.12.9.jar:na]
	...
	at org.apache.olingo.odata2.core.Dispatcher.dispatch(Dispatcher.java:92) ~[olingo-odata2-core-3.0.0-...]
	at org.apache.olingo.odata2.core.ODataRequestHandler.handle(ODataRequestHandler.java:131) ~[olingo-odata2-core-3.0.0-...]
	... 78 common frames omitted
Caused by: java.lang.NullPointerException: null
	at org.apache.olingo.odata2.core.commons.Encoder.encodeInternal(Encoder.java:101) ~[olingo-odata2-core-3.0.0-...]
	at org.apache.olingo.odata2.core.commons.Encoder.encode(Encoder.java:41) ~[olingo-odata2-core-3.0.0-...]
	at org.apache.olingo.odata2.core.ep.producer.AtomEntryEntityProducer.createEntryKey(AtomEntryEntityProducer.java:741) ~[olingo-odata2-core-3.0.0-...]
	at org.apache.olingo.odata2.core.ep.producer.AtomEntryEntityProducer.createSelfLink(AtomEntryEntityProducer.java:702) ~[olingo-odata2-core-3.0.0-...]
	at org.apache.olingo.odata2.core.ep.producer.JsonEntryEntityProducer.writeMetadata(JsonEntryEntityProducer.java:363) ~[olingo-odata2-core-3.0.0-...]
	at org.apache.olingo.odata2.core.ep.producer.JsonEntryEntityProducer.append(JsonEntryEntityProducer.java:86) ~[olingo-odata2-core-3.0.0-...]
	at org.apache.olingo.odata2.core.ep.JsonEntityProvider.writeEntry(JsonEntityProvider.java:142) ~[olingo-odata2-core-3.0.0-...]
	... 88 common frames omitted
```

**Expected behavior**
According to the specification https://www.odata.org/documentation/odata-version-2-0/operations/ a 204 should be returned after executing a PUT request (we assume the same is true for MERGE/PATCH requests, however the specification is fuzzy here).
Currently the implementation tries to return the entity (in PUT and MERGE/PATCH calls) and this results in the NPE.

--- a/modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/builder/expression/SQLExpressionUpdate.java
+++ b/modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/builder/expression/SQLExpressionUpdate.java
@@ -31,12 +31,12 @@ import static org.eclipse.dirigible.engine.odata2.sql.utils.OData2Utils.fqn;
 public class SQLExpressionUpdate implements SQLExpression {
 
 	private final SQLQuery query;
-	private EdmEntityType target;
-	private ODataEntry requestEntry;
+	private final EdmEntityType target;
+	private final ODataEntry requestEntry;
 	private final List<String> nonKeyColumnNames = new ArrayList<>();
 	private final List<Object> queryData = new ArrayList<>();
 	private final List<EdmProperty> queryOdataProperties = new ArrayList<>();
-	private Map<String, Object> uriKeyProperties;
+	private final Map<String, Object> uriKeyProperties;
 
 	public SQLExpressionUpdate(final SQLQuery parent, EdmEntityType target, ODataEntry requestEntry, Map<String, Object> uriKeys) {
 		this.query = parent;
@@ -47,18 +47,15 @@ public class SQLExpressionUpdate implements SQLExpression {
 
 	@Override
 	public String evaluate(final SQLContext context, final ExpressionType type) throws EdmException {
-		switch (type) {
-		case TABLE:
+		if (type == ExpressionType.TABLE) {
 			return buildStatement(context);
-		default:
-			throw new OData2Exception("Unable to evaluate the SQLSelect to type " + type,
-					HttpStatusCodes.INTERNAL_SERVER_ERROR);
 		}
+		throw new OData2Exception("Unable to evaluate the SQLSelect to type " + type, HttpStatusCodes.INTERNAL_SERVER_ERROR);
 	}
 
 	@Override
 	public boolean isEmpty() throws EdmException {
-		return (nonKeyColumnNames.size() == 0 ? true : false);
+		return (nonKeyColumnNames.size() == 0);
 	}
 
 	public EdmStructuralType getTarget() {
@@ -68,7 +65,6 @@ public class SQLExpressionUpdate implements SQLExpression {
 	public SQLQuery build() throws ODataException {
 		query.grantTableAliasForStructuralTypeInQuery(target);
 		Map<String, Object> entryValues = requestEntry.getProperties();
-
 
 		for (EdmProperty property : EdmUtils.getProperties(target)) { //we iterate first the own properties of the type
 			if (entryValues.containsKey(property.getName()) && !isKeyProperty(target, property)) {
@@ -129,7 +125,7 @@ public class SQLExpressionUpdate implements SQLExpression {
 
 	protected boolean isUpdateTarget(final EdmStructuralType target) {
 		// always select the entity target
-		return fqn(query.getUpdateExpression().getTarget()).equals(fqn(target)) ? true : false;
+		return fqn(query.getUpdateExpression().getTarget()).equals(fqn(target));
 	}
 
 	protected String buildColumnList(final SQLContext context) {

--- a/modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/processor/AbstractSQLProcessor.java
+++ b/modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/processor/AbstractSQLProcessor.java
@@ -11,6 +11,7 @@
  */
 package org.eclipse.dirigible.engine.odata2.sql.processor;
 
+import org.apache.olingo.odata2.api.commons.HttpStatusCodes;
 import org.apache.olingo.odata2.api.commons.InlineCount;
 import org.apache.olingo.odata2.api.edm.*;
 import org.apache.olingo.odata2.api.ep.EntityProvider;
@@ -184,7 +185,6 @@ public abstract class AbstractSQLProcessor extends ODataSingleProcessor implemen
 			LOG.error("Unable to serve request", e);
 			throw new ODataException(e);
 		}
-		
 		ODataResponse response = OData2Utils.writeEntryWithExpand(getContext(), (UriInfo) uriInfo, resultEntity, contentType);
 		return response;
 	}
@@ -622,14 +622,7 @@ public abstract class AbstractSQLProcessor extends ODataSingleProcessor implemen
 			LOG.error("Unable to serve request", e);
 			throw new ODataException(e);
 		}
-
-		ODataContext context = getContext();
-		EntityProviderWriteProperties writeProperties = EntityProviderWriteProperties
-				.serviceRoot(context.getPathInfo().getServiceRoot()).expandSelectTree(entry.getExpandSelectTree())
-				.build();
-		final ODataResponse response = EntityProvider.writeEntry(contentType, targetEntitySet, entry.getProperties(),
-				writeProperties);
-		
+		ODataResponse response = ODataResponse.newBuilder().status(HttpStatusCodes.NO_CONTENT).build();
 		this.odata2EventHandler.afterUpdateEntity(uriInfo, requestContentType, merge, contentType, entry);
 		return response;
 	}

--- a/modules/odata/odata-core/src/test/java/org/eclipse/dirigible/engine/odata2/sql/ODataSQLProcessorTest.java
+++ b/modules/odata/odata-core/src/test/java/org/eclipse/dirigible/engine/odata2/sql/ODataSQLProcessorTest.java
@@ -436,6 +436,58 @@ public class ODataSQLProcessorTest {
         assertEquals(50000.0d, properties.get("Price"));
     }
 
+    @Test
+    public void testPatchEntity() throws Exception {
+        Response existingCar = OData2RequestBuilder.createRequest(sf) //
+                .segments("Cars('639cac17-4cfd-4d94-b5d0-111fd5488423')") //
+                .accept("application/json").executeRequest(GET);
+        assertEquals(200, existingCar.getStatus());
+
+        String content = "{" //
+                + "  \"d\": {" //
+                + "    \"__metadata\": {" //
+                + "      \"type\": \"org.eclipse.dirigible.engine.odata2.sql.entities.Car\"" //
+                + "    }," //
+                + "    \"Price\": 50001.0" //
+                + " }" + "}";
+
+        Response response = modifyingRequestBuilder(sf, content)//
+                .segments("Cars('639cac17-4cfd-4d94-b5d0-111fd5488423')") //
+                .accept("application/json")//
+                .content(content)//
+                .param("content-type", "application/json")//
+                .contentSize(content.length()).executeRequest(PATCH);
+        assertEquals(204, response.getStatus());
+
+        existingCar = OData2RequestBuilder.createRequest(sf) //
+                .segments("Cars('639cac17-4cfd-4d94-b5d0-111fd5488423')") //
+                .accept("application/json").executeRequest(GET);
+        assertEquals(200, existingCar.getStatus());
+        ODataEntry resultEntry = retrieveODataEntry(existingCar, "Cars");
+        Map<String, Object> properties = resultEntry.getProperties();
+        assertEquals("BMW", properties.get("Make"));
+        assertEquals("530e", properties.get("Model"));
+        assertEquals(2021, properties.get("Year"));
+        assertEquals(50001.0d, properties.get("Price"));
+    }
+
+    @Test
+    public void testPatchEntityNonExistingId() throws Exception {
+        String content = "{" //
+                + "  \"d\": {" //
+                + "    \"__metadata\": {" //
+                + "      \"type\": \"org.eclipse.dirigible.engine.odata2.sql.entities.Car\"" //
+                + "    }," //
+                + "    \"Price\": 50001.0" //
+                + " }" + "}";
+        Response response = modifyingRequestBuilder(sf, content)//
+                .segments("Cars('88888888-8888-8888-8888-888888888888')") //
+                .accept("application/json")//
+                .content(content)//
+                .param("content-type", "application/json")//
+                .contentSize(content.length()).executeRequest(PATCH);
+        assertEquals(404, response.getStatus());
+    }
 
     @Test
     public void testPutEntityWithFilterNotAllowed() throws Exception {

